# Barca FC Notebook
Spanish League Data from 1929-2022

How many Columns

In [2]:
import pandas as pd

spain_df = pd.read_csv("data/spain.csv")
spain_df.columns

Index(['Date', 'Season', 'home', 'visitor', 'HT', 'FT', 'hgoal', 'vgoal',
       'tier', 'round', 'group', 'notes'],
      dtype='object')

How many Rows:

In [2]:
spain_df.shape

(26195, 12)

Basic Info:

Gives data on count totals for each column and the data type.  

In [3]:
spain_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26195 entries, 0 to 26194
Data columns (total 12 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Date     26195 non-null  object
 1   Season   26195 non-null  int64 
 2   home     26195 non-null  object
 3   visitor  26195 non-null  object
 4   HT       23155 non-null  object
 5   FT       26195 non-null  object
 6   hgoal    26195 non-null  int64 
 7   vgoal    26195 non-null  int64 
 8   tier     26195 non-null  int64 
 9   round    26195 non-null  object
 10  group    90 non-null     object
 11  notes    1 non-null      object
dtypes: int64(4), object(8)
memory usage: 2.4+ MB


In [4]:
spain_df.isnull().sum().sort_values(ascending=False)

notes      26194
group      26105
HT          3040
Date           0
Season         0
home           0
visitor        0
FT             0
hgoal          0
vgoal          0
tier           0
round          0
dtype: int64

This info tells us that there are scores for 26,195 games with 3,040 games missing halftime scores.

There are 90 games with "group" data. These are likely tournament games?

to preview the data we will first use the head function to view the first couple rows

In [6]:
spain_df.head(10)

Date  Season                home             visitor   HT   FT  \
0  1929-02-10    1928     Arenas de Getxo     Atletico Madrid  0-2  2-3   
1  1929-02-10    1928  Espanyol Barcelona          Real Union  1-0  3-2   
2  1929-02-10    1928         Real Madrid           CE Europa  0-0  5-0   
3  1929-02-10    1928       Real Sociedad     Athletic Bilbao  1-1  1-1   
4  1929-02-12    1928    Racing Santander        FC Barcelona  0-0  0-2   
5  1929-02-17    1928        FC Barcelona         Real Madrid  0-1  1-2   
6  1929-02-17    1928     Athletic Bilbao  Espanyol Barcelona  3-0  9-0   
7  1929-02-17    1928     Atletico Madrid       Real Sociedad  0-3  0-3   
8  1929-02-17    1928          Real Union    Racing Santander  0-1  3-1   
9  1929-02-17    1928           CE Europa     Arenas de Getxo  2-0  5-2   

   hgoal  vgoal  tier   round group notes  
0      2      3     1  league   NaN   NaN  
1      3      2     1  league   NaN   NaN  
2      5      0     1  league   NaN   NaN  
3      1      1     1  league   NaN   NaN  
4      0      2     1  league   NaN   NaN  
5      1      2     1  league   NaN   NaN  
6      9      0     1  league   NaN   NaN  
7      0      3     1  league   NaN   NaN  
8      3      1     1  league   NaN   NaN  
9      5      2     1  league   NaN   NaN

We can also just call the dataframe directly.

In [3]:
spain_df

Date  Season                home             visitor   HT   FT  \
0      1929-02-10    1928     Arenas de Getxo     Atletico Madrid  0-2  2-3   
1      1929-02-10    1928  Espanyol Barcelona          Real Union  1-0  3-2   
2      1929-02-10    1928         Real Madrid           CE Europa  0-0  5-0   
3      1929-02-10    1928       Real Sociedad     Athletic Bilbao  1-1  1-1   
4      1929-02-12    1928    Racing Santander        FC Barcelona  0-0  0-2   
...           ...     ...                 ...                 ...  ...  ...   
26190  2022-05-22    2021          Granada CF  Espanyol Barcelona  NaN  0-0   
26191  2022-05-22    2021          CA Osasuna        RCD Mallorca  NaN  0-2   
26192  2022-05-22    2021        FC Barcelona       Villarreal CF  NaN  0-2   
26193  2022-05-22    2021          Sevilla FC     Athletic Bilbao  NaN  1-0   
26194  2022-05-22    2021       Real Sociedad     Atletico Madrid  NaN  1-2   

       hgoal  vgoal  tier   round group notes  
0          2      3     1  league   NaN   NaN  
1          3      2     1  league   NaN   NaN  
2          5      0     1  league   NaN   NaN  
3          1      1     1  league   NaN   NaN  
4          0      2     1  league   NaN   NaN  
...      ...    ...   ...     ...   ...   ...  
26190      0      0     1  league   NaN   NaN  
26191      0      2     1  league   NaN   NaN  
26192      0      2     1  league   NaN   NaN  
26193      1      0     1  league   NaN   NaN  
26194      1      2     1  league   NaN   NaN  

[26195 rows x 12 columns]

This shows us a preview of the data with both the first and last rows.

It looks like we have data up to March 22, 2022. As a Barcelona fan, it was a sad day to end a rough season.